python/tank/util/shotgun.py

```diff
@@ -151,67 +158,75 @@ def __get_sg_config_data(shotgun_cfg_path, user="default"):
+
+    # now check if there is a studio level override hook which want to refine these settings
     sg_hook_path = os.path.join(__get_api_core_config_location(), constants.STUDIO_HOOK_SG_CONNECTION_SETTINGS)
-    
+
     if os.path.exists(sg_hook_path):
         # custom hook is available!
-        config_data = hook.execute_hook(sg_hook_path, 
-                                        parent=None, 
-                                        config_data=config_data, 
-                                        user=user, 
+        config_data = hook.execute_hook(sg_hook_path,
+                                        parent=None,
+                                        config_data=config_data,
+                                        user=user,
                                         cfg_path=shotgun_cfg_path)
-        
-    # validate the config data to ensure all fields are present
-    if "host" not in config_data:
-        raise TankError("Missing required field 'host' in config '%s'" % shotgun_cfg_path)
-    if "api_script" not in config_data:
-        raise TankError("Missing required field 'api_script' in config '%s'" % shotgun_cfg_path)
-    if "api_key" not in config_data:
-        raise TankError("Missing required field 'api_key' in config '%s'" % shotgun_cfg_path)
-    
+
+    def _raise_missing_key(key):
+        raise TankError(
+            "Missing required field '%s' in config '%s' for script user authentication." % (key, shotgun_cfg_path)
+        )
+
+    if not config_data.get("host"):
+        _raise_missing_key("host")
+
+    # The script authentication credentials need to be complete in order to work. They can be completely
+    # omitted or fully specified, but not halfway configured.
+    if config_data.get("api_script") and not config_data.get("api_key"):
+        _raise_missing_key("api_key")
+    if not config_data.get("api_script") and config_data.get("api_key"):
+        _raise_missing_key("api_script")
+
     return config_data
 
-def __create_sg_connection(shotgun_cfg_path, evaluate_script_user, user="default"):
-    """
-    Creates a standard toolkit shotgun connection.
 
-    :param shotgun_cfg_path: path to a configuration file to read settings from
-    :param evaluate_script_user: if True, the id of the script user will be 
-                                 looked up and returned.
-    :param user: If a multi-user config is used, this is the user to create the connection for.
-    
-    :returns: tuple with (sg_api_instance, script_user_dict) where script_user_dict is None if
-              evaluate_script_user is False else a dictionary with type and id keys. 
+def __create_sg_connection(config_data=None):
     """
+    Creates a standard Toolkit shotgun connection.
 
-    # get connection parameters
-    config_data = __get_sg_config_data(shotgun_cfg_path, user)
+    :param shotgun_cfg_path: Configuration data. If None, the authentication module will be responsible
```

Comment:
Parameter is called `config_data`, not `shotgun_cfg_path`. Also would be great to know more details around structure is expected for this data.